klaxon/src/main/kotlin/com/beust/klaxon/Lexer.kt

```diff
@@ -115,13 +115,21 @@ class Lexer(passedReader: Reader, val lenient: Boolean = false): Iterator<Token>
                             'r' -> currentValue.append("\r")
                             't' -> currentValue.append("\t")
                             'u' -> {
-                                val unicodeChar = StringBuilder(4)
-                                    .append(nextChar())
-                                    .append(nextChar())
-                                    .append(nextChar())
-                                    .append(nextChar())
-
-                                val intValue = java.lang.Integer.parseInt(unicodeChar.toString(), 16)
+                                val unicodeChar = try {
+                                    StringBuilder(4)
+                                        .append(nextChar())
+                                        .append(nextChar())
+                                        .append(nextChar())
+                                        .append(nextChar())
+                                } catch(_: IllegalStateException) {
+                                    throw KlaxonException("EOF reached in unicode char")
```

Comment:
Maybe add the unicodeChar in the exception message.

Reply:
I'm now adding the current state of the `StringBuilder` to the exception message.